api/v1alpha1/serverproviderprovisionpoolpolicy_types.go

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// PoolPolicy defines the proactive provisioning policy for idle launcher pods.
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:path=serverproviderprovisionpoolpolicies,scope=Namespaced
+
+type ServerProviderProvisionPoolPolicy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   PoolPolicySpec   `json:"spec,omitempty"`
+	Status PoolPolicyStatus `json:"status,omitempty"`
+}
+
+// PoolPolicySpec defines the node-level idle pool configuration.
+type PoolPolicySpec struct {
+	// PoolsPerNode defines idle launcher counts per node type.
+	PoolsPerNode []NodePoolSpec `json:"poolsPerNode"`
+}
+
+// NodePoolSpec defines launcher count for a class of nodes.
+type NodePoolSpec struct {
```

Comment:
What if multiple NodePoolSpec match one Node? There are two levels of multiplicity for NodePoolSpec in the system: there can be multiple policy objects, and each policy object can have multiple NodePoolSpec.

Note that because a NodeSelector specifies only _lower_ bounds, it is very easy for multiple NodePoolSpec to match one Node.

Reply:
Agree.Same comment  with below. 
I will change the PoolPolicy CRD to cluster scope, and  within the reconciler logic, validate the actual selected nodes. 
If more than one PoolPolicy CR is found targeting the same node, we should raise an error or warning.

In the RapidLLMServe project, we've relied on certain conventions to ensure that only one such definition exists per node.